tests/queries/0_stateless/02980_dist_insert_readonly_replica.sql.j2

```diff
@@ -0,0 +1,84 @@
+-- Tags: no-parallel, no-fasttest
+-- Tag no-parallel - due to static databases
+-- Tag no-fasttest - S3 is required
+
+drop database if exists shard_0;
+drop database if exists shard_1;
+create database shard_0;
+create database shard_1;
+
+drop table if exists dist;
+drop table if exists dist_batch;
+drop table if exists shard_0.data;
+drop table if exists shard_1.data;
+
+-- normal replica
+create table shard_0.data (key Int) engine=ReplicatedMergeTree('/tables/{database}/data', 'write') order by key;
+-- create read-only replica
+create table shard_1.data (key Int) engine=ReplicatedMergeTree('/tables/{database}/data', 'read') order by key;
+
+-- broke replica and leave it read-only
+detach table shard_1.data;
+insert into system.zookeeper (path, name, value) values ('/tables/shard_1/data/replicas/read', 'columns', '');
+attach table shard_1.data;
+select database||'.'||table, is_readonly from system.replicas where database in ('shard_0', 'shard_1') and table = 'data';
+
+create table dist as shard_0.data engine=Distributed('test_cluster_two_replicas_different_databases_internal_replication', '', 'data');
+system stop distributed sends dist;
+
+create table dist_batch as shard_0.data engine=Distributed('test_cluster_two_replicas_different_databases_internal_replication', '', 'data') settings background_insert_batch=1;
+system stop distributed sends dist_batch;
+
+create table dist_single_no_internal_replication as shard_0.data engine=Distributed('default', 'shard_1', 'data');
```

Comment:
Can we name table with `read_only` in it. So, it'd clear from name if we refer to read-only table